src/main/java/org/openRealmOfStars/gui/util/GuiStatics.java

```diff
@@ -71,345 +75,172 @@ private GuiStatics() {
+      .getDataUrl("resources/fonts/Cubellan_v_0_7/Cubellan_SC.ttf")
+      .orElse(null);
+  /** Squarion hinted Font URL */
+  private static final URL SQUARION_HINTED_FONT_URL = DataSources
+      .getDataUrl("resources/fonts/Squarion/hinted-Squarion.ttf")
+      .orElse(null);
+
+  /** The Font cache. It caches fonts... End of mandatory Javadoc comment. */
+  private static HashMap<String, Font> fontCache = new HashMap<>();
+
+  /**
+   * Load a TrueType font in it's basic form.
+   * @param fontUrl URL to Font
+   * @return Loaded Font or empty
+   */
+  private static Optional<Font> loadTrueTypeFont(
+      final URL fontUrl) {
+    try (InputStream is = fontUrl.openStream()) {
+      return Optional.of(Font.createFont(Font.TRUETYPE_FONT, is));
+    } catch (IOException | FontFormatException e) {
+      ErrorLogger.log("Font loading error:" + e.getMessage());
+      return Optional.empty();
+    }
+  }
 
   /**
-   * Regular cubellan font but smaller
+   * Creates a "Font ID" from input. Used for Font caching identifiers.
+   * @param fontResource Font URL as String
+   * @param baseSize base size of the font
+   * @return ID of Font for fontCache
    */
-  private static Font fontCubellanSmall;
+  private static String buildFontId(final String fontResource,
+      final float baseSize) {
+    return fontResource.toString() + "SIZE:" + baseSize;
+  }
 
   /**
-   * Regular cubellan font but smaller and larger
+   * Lazy-load a font of specified base size and store to cache
+   * @param fontUrl Font URL
+   * @param baseSize base size of the font
+   * @return Font or empty if loading failed
    */
-  private static Font fontCubellanSmallLarger;
+  private static Optional<Font> getFontFromCache(final URL fontUrl,
+      final float baseSize) {
+    if (fontUrl == null) {
+      return Optional.empty();
+    }
+
+    final var fontOpt = loadTrueTypeFont(fontUrl);
+    final var fontSize = baseSize * fontScalingFactor;
+
+    if (fontOpt.isPresent()) {
+      final var tmpFont = fontOpt.get().deriveFont(fontSize);
+      final var fontId = buildFontId(fontUrl.toExternalForm(), baseSize);
+      fontCache.put(fontId, tmpFont);
+      return Optional.of(tmpFont);
+    }
+    return Optional.empty();
+  }
 
   /**
    * Get Regular Cubellan font very small
    * @return Cubellan font
    */
   public static Font getFontCubellanVerySmall() {
-    if (fontVerySmallCubellan == null) {
-      try (InputStream is = Tiles.class
-          .getResource("/resources/fonts/Cubellan_v_0_7/Cubellan.ttf")
-          .openStream()) {
-        fontVerySmallCubellan = Font.createFont(Font.TRUETYPE_FONT, is);
-        fontVerySmallCubellan = fontVerySmallCubellan.deriveFont(9F);
-      } catch (IOException | FontFormatException e) {
-        ErrorLogger.log("Error:" + e.getMessage());
-        return FONT_SMALL;
-      }
-    }
-    return fontVerySmallCubellan;
+    return getFontFromCache(CUBELLAN_FONT_URL, 9).orElse(FONT_SMALL);
```

Comment:
This font is only used in star map view drawing when painting the sun's name. Since it is tile based there font cannot be scaled.  Should we have some way to have fixed font?